src/main/java/com/suse/salt/netapi/datatypes/target/Grains.java

```diff
@@ -3,45 +3,43 @@
 /**
  * Matcher based on salt grains
  */
-public class Grains implements Target<String> {
-
-    private final String grain;
-    private final String value;
-    private final String target;
+public class Grains extends DictionaryTarget implements Target<String> {
 
     /**
      * Creates a grains matcher
      *
      * @param grain the grain name
      * @param value the value to match
+     *
+     * @deprecated
+     * Use {@link #Grains(DictionaryTargetExpression)} instead.
      */
+    @Deprecated
     public Grains(String grain, String value) {
```

Comment:
Having a constructor like this is more convenient to use then having to create a `DictionaryTargetExpression`.